scripts/add_class

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+
+# Usage:
+# add_class.py [-o] COMPONENT QUALIFIED-NAME
+# Options:
+# -o   Header-only (do not add .cpp file)
+
+# Usage example:
+# - Header, class, and test: add_class.py libvast vast::system::foo
+# - Header and test: add_class.py -o libvast vast::system::foo
+
+import argparse
+
+copyright="""/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+"""
+
+header_tpl="""
+#pragma once
+
+namespace %(namespace)s {
+
+class %(class)s {
+public:
+  %(class)s();
+  ~%(class)s();
```

Comment:
For the majority of types, we don't have a destructor.

I would even say that we don't need to make the assumption that we have a no-argument constructor that's not default.

Reply:
> For the majority of types, we don't have a destructor.

Maybe, but I'm sure many should have (because of weak vtables, for example).

I find that a minimal scaffold like this usually saves some typing. Especially a scaffold for a constructor where I only need to put in my arguments. I don't have strong feelings regarding the template, though. Do you prefer the scaffold without any members?

Reply:
> I don't have strong feelings regarding the template, though. Do you prefer the scaffold without any members?

Me neither 😀. Keep or remove 'em--your call.

Reply:
I'd say we keep them. Deleting if you don't need them is faster than typing them if you do.